cli/src/main/java/org/infinispan/cli/commands/rest/Connector.java

```diff
@@ -0,0 +1,216 @@
+package org.infinispan.cli.commands.rest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.aesh.command.CommandDefinition;
+import org.aesh.command.CommandException;
+import org.aesh.command.CommandResult;
+import org.aesh.command.GroupCommandDefinition;
+import org.aesh.command.option.Argument;
+import org.aesh.command.option.Option;
+import org.aesh.command.option.OptionList;
+import org.infinispan.cli.activators.ConnectionActivator;
+import org.infinispan.cli.commands.CliCommand;
+import org.infinispan.cli.completers.ConnectorCompleter;
+import org.infinispan.cli.completers.IpFilterRuleCompleter;
+import org.infinispan.cli.impl.ContextAwareCommandInvocation;
+import org.infinispan.cli.logging.Messages;
+import org.infinispan.cli.resources.Resource;
+import org.infinispan.client.rest.IpFilterRule;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 12.0
+ **/
+@GroupCommandDefinition(name = Connector.CMD, description = "Performs operations on protocol connectors", activator = ConnectionActivator.class, groupCommands = {Connector.Ls.class, Connector.Describe.class, Connector.Start.class, Connector.Stop.class, Connector.IpFilter.class})
+public class Connector extends CliCommand {
```

Comment:
It seems like we have a discrepancy between our xml and CLI.

The CLI allows ip-filters  defined per connectors to be updated, however it's not possible to update those defined globally under `<endpoints><ip-filter>`

Should we have a root `endpoints` command/rest endpoint to allow this level of configuration?

So the CLI would be as follows:
```
$ cli.sh endpoints status hotrod
$ cli.sh endpoints stop hotrod
$ cli.sh endpoints start hotrod
$ cli.sh endpoints ipfilter set .... # global filters
$ cli.sh endpoints ipfilter --connector=hotrod set .... # hotrod specific filters
```

I think this would need the rest endpoint to be updated to the following to allow for this flexibility:

```
/v2/server/endpoints/ip-filter # global
/v2/server/endpoints/connectors/{connector}
/v2/server/endpoints/connectors/{connector}/ip-filter # connector specific
```

This would also have the advantage of allowing configuration at the `endpoints` level to be exposed in the future.

Reply:
Internal connector naming is what matters. In a default server:
`["REST-rest-default","HotRod-hotrod-default","endpoint-default"]`

```
[bespin-13557@cluster//containers/default]> server connector describe endpoint-default 
{
  "name" : "endpoint-default",
  "ip-filter-rules" : [  ],
  "host" : "localhost",
  "port" : 11222,
  "local-connections" : 1,
  "global-connections" : 1,
  "io-threads" : 16,
  "pending-tasks" : 0,
  "total-bytes-read" : 0,
  "total-bytes-written" : 0,
  "send-buffer-size" : 0,
  "receive-buffer-size" : 0,
  "enabled" : true
}
```

Reply:
Argh I see. For me it's counter-intuitive for the "endpoint-default" to be accessed via the "connector" field in the CLI, when `<ip-filter>` is a child of `<endpoint>` in the xml.

Also utilising the REST endpoint of `/v2/server/connectors` opposed to `/v2/server/endpoints/` limits us if we add any additional child elements under `<endpoints>` that we want to expose in the future.

Reply:
Well, the single port endpoint *is* a connector that multiplexes other connectors. Also, a hotrod connector assigned to a different socket binding, i.e. not part of a single port, is conceptually a top-level connector, even though it needs to be "housed" inside the endpoint element in the XML (something I'm not particularly fond of)